File: WarehouseApplication.Server/Models/Supplier.cs
using System.ComponentModel.DataAnnotations;

namespace WarehouseApplication.Server.Models
{
    public class Supplier
    {
        [Key]
        public int SupplierID { get; set; }
        public string SupplierName { get; set; }

        public int AddressID { get; set; }
        public Address Address { get; set; }

        public ICollection<DeliveryDocument>? Documents { get; set; }
    }
}

File: warehouseapplication.client/src/app/models/supplier.ts
import { DeliveryDocument } from './delivery-document';

export interface Supplier {
  supplierID: number;
  supplierName: string;
  address: {
    street: string;
    city: string;
    country: string;
    zipcode: string;
  };
  /*documents?: DeliveryDocument[];*/
}
